extensions/ipc/internal.m

```diff
@@ -129,7 +129,11 @@ static int ipc_remotePort(lua_State *L) {
     NSString *portName = [skin toNSObjectAtIndex:1] ;
 
     HSIPCMessagePort *port = [[HSIPCMessagePort alloc] init] ;
-    if (port) port.messagePort = CFMessagePortCreateRemote(NULL, (__bridge CFStringRef)portName) ;
+    if (port) {
+        CFMessagePortRef portRef = CFMessagePortCreateRemote(NULL, (__bridge CFStringRef)portName) ;
+        port.messagePort = portRef;
+        CFRelease(portRef);
+    }
```

Comment:
`HSIPCMessagePort`'s `messagePort` property is implicitly `strong`, so it will retain a reference. If we don't `CFRelease()` here we're leaking.

Reply:
Are core foundation type properties default strong? I always thought strong was limited to Objective-C objects and CF objects were treated like any other pointer.

Explicitly making it strong and compiling results in:

~~~
hammerspoon/extensions/ipc/internal.m:14:1: error: property with 'retain (or strong)' attribute must be of object type
@property (strong) CFMessagePortRef   messagePort ;
~~~

Reply:
Huh, that’s a really good point. I’m not sure what to do about this one now.

Reply:
I did come across a discussion in the clang forums that talks about casting CF types when defining them as properties to NSObject so that ARC could handle memory management within the auto-synthesized getter and setter functions -- they were clear, though, that it *only* affected the code within the synthesized functions and use of the ivar within the class (i.e. accessing it as `_var` instead of as `self.var`) still required us to handle proper memory management... I'll see if I can find links to the official documentation.

Probably the first thing I need to do is get my setup caught up to yours, pull out a few of my more experimental modules that I'm not really working on at the moment, and run with the same analyzers and debug options within XCode so I can see the same reports you're getting in real time and in context.  Tonight is pretty busy, but I'll shoot for being in a position to compare apples to apples tomorrow.

Reply:
Thinking about this some more, I think Xcode's analysis error may well be simply that it doesn't understand what's happening with the memory - it sees us assign it to a property, probably knows that the property won't retain, and then the end of the function arrives. It can't track it across to the `__gc` functions where the memory is released, so there probably isn't actually a leak here, just an implicit behaviour. I'm not super worried about that, but it would be nice if we could find a way to hint to the compiler that we know what we're doing :)